contrib/HeliconFocus.lua

```diff
@@ -0,0 +1,491 @@
+    end
+
+    if dt.control.ending or not job.valid then
+      dt.print_log( _"exporting images canceled!")
+      inputfilelist:close()
+      df.rmdir( tempdirname )
+      return
+    end
+
+    -- update progress_bar
+    job.percent = i / #images
+
+    -- sleep for a short moment to give stop_job callback function a chance to run
+    dt.control.sleep(10)    
+  end
+  
+  -- stop job and remove progress_bar from ui, but only if not alreay canceled
+  if(job.valid) then
+    job.valid = false
+  end
+
+  inputfilelist:close()
+
+  local helicon_commandline = df.sanitize_filename( dt.preferences.read( mod, "HeliconFocusExe", "string" ) )
+
+  helicon_commandline = helicon_commandline .. ' -i ' .. df.sanitize_filename( inputfilelistname )
+                                            .. ' -o ' .. df.sanitize_filename( outputfilelistname )
+                                            .. ' --preferred-output-path ' .. firstimagepath
+
+  dt.print_log( 'preferred output path: '..firstimagepath )
+  dt.print_log( 'temp directory: '..tempdirname )
+  dt.print_log( 'outptufilelist: '..outputfilelistname )
+  dt.print_log( 'commandline: '..helicon_commandline )
+
+  dt.print_log( 'starting helicon' )
+  resp = dsys.external_command( helicon_commandline )
+  dt.print_log( 'helicon returned '..tostring( resp ) )
+
+  if resp ~= 0 then
+    dt.print( _'could not start HeliconFocus application' )
+    df.rmdir( tempdirname )
+    return
+  end
+
+  stackedimages = reimport_images( outputfilelistname )
+  if #stackedimages == 0 then
+    dt.print( _"no image to import" )
+    df.rmdir( tempdirname )
+    return
+  end
+
+  -- group source images:
+  if GUI.optionwidgets.group.value ~= false then
+    for _,imagetogroup in pairs( images_to_group ) do
+      imagetogroup:group_with( images[ 1 ] )
+    end
+  end
+
+  dt.print_log( 'group with: '..images[ 1 ].filename )
+  -- group stacked images and add tags:
+  for _,stackedimage in pairs( stackedimages ) do
+    dt.print_log( 'stacking: '..stackedimage.filename.. ' to '..images[ 1 ].filename )
+
+    -- group output images
+    if GUI.optionwidgets.group.value ~= false then
+      stackedimage:group_with( images[ 1 ] )
+    end
+
+    -- copy tags:
+    if GUI.optionwidgets.copy_tags.value ~= false then
+      insert_tags( stackedimage, all_tags )
+    end
+
+    add_additional_tags( stackedimage )
+
+  end
+
+  stackedimages[ 1 ]:make_group_leader()
+
+  -- cleanup:
+  df.rmdir( tempdirname )
+
+end
+
+GUI.optionwidgets.use_original = dt.new_widget('check_button' ) {
+  label = _('use original RAW'),
+  value = false,
+  tooltip = _('stack original RAW images instead of exported'),
+  clicked_callback = function(self)
+    dt.print_log( "use_original: "..tostring( self.value ) )
+  end,
+  reset_callback = function(self)
+    self.value = false
+  end
+}
+
+GUI.optionwidgets.group = dt.new_widget('check_button') {
+  label = _('group'),
+  value = false,
+  tooltip = _('group stacked images'),

```

Comment:
This desciption is missing the information that the result  image is also part of the created group.
Suggestion: "group selected source images and imported results together"